applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -443,18 +452,26 @@ void DistanceModificationProcess::DeactivateFullNegativeElements() {
     mrModelPart.GetCommunicator().AssembleNonHistoricalData(EMBEDDED_IS_ACTIVE);
 
     // Set to zero and fix the DOFs in the remaining inactive nodes
-    #pragma omp parallel for
-    for (int i_node = 0; i_node < static_cast<int>(rNodes.size()); ++i_node){
-        ModelPart::NodesContainerType::iterator it_node = rNodes.begin() + i_node;
-        if (it_node->GetValue(EMBEDDED_IS_ACTIVE) == 0){
-            // Fix the nodal DOFs
-            it_node->Fix(PRESSURE);
-            it_node->Fix(VELOCITY_X);
-            it_node->Fix(VELOCITY_Y);
-            it_node->Fix(VELOCITY_Z);
-            // Set to zero the nodal DOFs
-            it_node->FastGetSolutionStepValue(PRESSURE) = 0.0;
-            it_node->FastGetSolutionStepValue(VELOCITY) = ZeroVector(3);
+    if ((mDoubleVariablesList.size() > 0.0) || (mComponentVariablesList.size() > 0.0)){
```

Comment:
In the check below I was doing the same as here, but here we are avoiding the nodal loop if both variable list have size()==0.0